db/migrate/20160323170320_create_subjects.rb

```diff
@@ -1,7 +1,7 @@
 class CreateSubjects < ActiveRecord::Migration
   def change
     create_table :subjects do |t|
-      t.string :name, null: false, index: true
+      t.string :name, null: false, index: true, uniqueness: true
```

Comment:
I don't know if there is an automatic uniqueness generator but `bin/rails g migration NameOfMIgration` then edit to add uniqueness constraint (which I also don't know how to do).